x-pack/plugins/ml/public/explorer/explorer.js

```diff
@@ -0,0 +1,294 @@
+      swimlaneOverall: PropTypes.object,
+      swimlaneViewByFieldName: PropTypes.string,
+      tableData: PropTypes.object,
+      viewByLoadedForTimeFormatted: PropTypes.any,
+      viewBySwimlaneOptions: PropTypes.array,
+    };
+
+    viewByChangeHandler = e => this.props.setSwimlaneViewBy(e.target.value);
+
+    render() {
+      const {
+        annotationsData,
+        anomalyChartRecords,
+        influencers,
+        intl,
+        hasResults,
+        jobs,
+        loading,
+        noInfluencersConfigured,
+        setSwimlaneSelectActive,
+        showViewBySwimlane,
+        swimlaneOverall,
+        swimlaneViewBy,
+        swimlaneViewByFieldName,
+        tableData,
+        viewByLoadedForTimeFormatted,
+        viewBySwimlaneOptions,
+      } = this.props;
+
+      if (loading === true) {
+        return (
+          <LoadingIndicator
+            label={intl.formatMessage({
+              id: 'xpack.ml.explorer.loadingLabel',
+              defaultMessage: 'Loading',
+            })}
+          />
+        );
+      }
+
+      if (jobs.length === 0) {
+        return <ExplorerNoJobsFound />;
+      }
+
+      if (jobs.length > 0 && hasResults === false) {
+        return <ExplorerNoResultsFound />;
+      }
+
+      const mainColumnWidthClassName = noInfluencersConfigured === true ? 'col-xs-12' : 'col-xs-10';
+      const mainColumnClasses = `column ${mainColumnWidthClassName}`;
+
+      return (
+        <div className="results-container">
+          {noInfluencersConfigured && (
+            <div className="no-influencers-warning">
+              <EuiIconTip
+                content={intl.formatMessage({
+                  id: 'xpack.ml.explorer.noConfiguredInfluencersTooltip',
+                  defaultMessage:
+                    'The Top Influencers list is hidden because no influencers have been configured for the selected jobs.',
+                })}
+                position="right"
+                type="iInCircle"
+              />
+            </div>
+          )}
+
+          {noInfluencersConfigured === false && (
+            <div className="column col-xs-2 euiText">
+              <span className="panel-title">
+                <FormattedMessage
+                  id="xpack.ml.explorer.topInfuencersTitle"
+                  defaultMessage="Top Influencers"
+                />
+                <InfluencersList influencers={influencers} />
+              </span>
+            </div>
+          )}
+
+          <div className={mainColumnClasses}>
+            <span className="panel-title euiText">
+              <FormattedMessage
+                id="xpack.ml.explorer.anomalyTimelineTitle"
+                defaultMessage="Anomaly timeline"
+              />
+            </span>
+
+            <div
+              className="ml-explorer-swimlane euiText"
+              onMouseEnter={() => setSwimlaneSelectActive(true)}
+              onMouseLeave={() => setSwimlaneSelectActive(false)}
+            >
```

Comment:
An option here to avoid creating these arrow functions in `render` might be to define handlers for each of these like `onMouseEnterHandler` which just always calls `setSwimLaneSelectActive` with `true`. Same for onMouseLeave.

Reply:
Done in 8d5bbf1a94.